### PR TITLE
Fix priority order language for tags in forms

### PR DESCRIPTION
**Pull request by inode64**
When we create or manage tags from frontend we have to check the param  'tag_list_language_filter' to show only corresponding tags.

### Summary of Changes
In the backend, we check only the language in the form because the manager can create content in any language.
In frontend, first of all, we use the language from the form and later we evaluate the param  'tag_list_language_filter' to select the correct language.

### Testing Instructions
I have checked that in the backend it works as good as it does now.

### Expected result
It shows only tags based on the param  'tag_list_language_filter' or the form in the frontend

### Actual result
It shows tags of all languages

### CODE REVIEW

File: libraries/src/Form/Field/TagField.php [hunk 1]
```diff
@@ -119,36 +119,37 @@ protected function getOptions()
...
+				$language = implode(',', $db->quote(explode(',', '*,' . $this->element['language'])));
+				$query->where($db->quoteName('a.language') . ' IN (' . $language . ')');
 			}
-			else
+			// Show all tags in the backend
+			elseif ($app->isClient('site'))
 			{
-				$language = $db->quote($this->element['language']);
+				// Second use current language from global configuration 
+				if ($lang == 'current_language')
```
Comment (C-Lodder) [inline libraries/src/Form/Field/TagField.php:142]: `==` >> `===`

File: libraries/src/Form/Field/TagField.php [hunk 2]
```diff
@@ -119,36 +119,37 @@ protected function getOptions()
...
+				{
+					$query->where($db->quoteName('a.language') . ' IN (' . $db->quote($tag) . ',' . $db->quote('*') . ')');
+				}
+				// Third use selected language from global configuration
+				elseif ($lang != 'all')
```
Comment (C-Lodder) [inline libraries/src/Form/Field/TagField.php:147]: `!=` >> `!==`

**Comment by ghost** [comment]
@inode64 can you please give Instructions how to test?

**Comment by inode64** [comment]
From the backend, It shows all tags, minus when in the form field is enabled language.
You change the option "Language Filter" from the backend in options Tags -> Item Selection
When selected to 'All' show all tags. and if selected to 'current' only it shows the tags in the current language selected in the frontend, and You selected a language, it only shows this language

**Comment by ghost** [comment]
@inode64 can you please update the Test Instructions on your first Comment so willing Tester don't have to search Thread?

**Comment by infograf768** [comment]
This should be tested on a multilingual site

**Comment by Giuse69** [comment]
I opened the related issue #17244 on backend and I am not convinced that in backend all tags must be shown: the admin user can freely select the language but then only selected language tags should be shown for consistency, otherwise the user can select one language and a tag of another language, the backend should prevent it.

**Comment by laoneo** [comment]
Sorry that it took so long to respond here. As this fix changes a behavior where Joomla users are used to it, it should be ported to version 4. Can you rebase your branch to 4.2 so we can reevaluate it for testing. In the meantime I'm closing this pr. When ready, please reopen again. Thanks for your contribution making Joomla better.

**Comment by laoneo** [comment]
Sorry that it took so long to respond here. As this fix changes a behavior where Joomla users are used to it, it should be ported to version 4. Can you rebase your branch to 4.2 so we can reevaluate it for testing. In the meantime I'm closing this pr. When ready, please reopen again. Thanks for your contribution making Joomla better.

**Pull request closed without merging**